### ISSUE TITLE
Progress on 1.13 support

### ISSUE THREAD

**Issue by mdcfe** [EssentialsX/Essentials#2223]
## Core

- [x] Bump version to 2.16.0
- [x] Compile against Bukkit ~~1.13~~ ~~1.13.1~~ 1.13.2
- [x] Materials support - ~~#2153~~, ~~#2204~~
  - [x] Abstract ItemDb for `items.csv` backwards-compatibility
  - [x] Remove IDs from `/itemdb`
  - [x] Remove item ID fallbacks in permission checks and config files
  - [x] New spawn egg materials
  - [x] Ensure item spawn blacklist and kits work - #2170 
  - [x] Ensure all item names are available - #2172
  - [x] Fix applying meta strings to items - #2204 [(comment 1)](https://github.com/EssentialsX/Essentials/pull/2204#issuecomment-430101827) [(comment 2)](https://github.com/EssentialsX/Essentials/pull/2204#issuecomment-431315029)
  - [x] Add enum convenience methods
  - [x] Improve item alias coverage
    - ~~Automatically download the latest `items.json` during build with [`maven-download-plugin`](https://github.com/maven-download-plugin/maven-download-plugin)~~ *no*
  - [x] Convert signs from item IDs to item names automatically
- [x] Update to Java 8 - ~~#1922~~
  - [x] Support Java 9 - ~~#1896~~
- ~~Evaluate command selectors support~~ - #2102 *not in 2.16.0*
- ~~Fix cross-world teleport safety~~ - #2121 *not until 2.16.1*
  - ~~Async teleportation/teleport safety (requires Paper API)~~ *not until 2.16.1*
- [x] Support new mobs

## Modules
- [x] Update AntiBuild/Protect to use item names - #2238

**Comment by jhammerberg** [comment]
I found on my server that /spawner seems to be saying "Target block must be a spawner." when im looking at a spawner. maybe add that to the list.

**Comment by mdcfe** [comment]
@Warp8 Please open a separate issue for your bug report.

**Comment by mdcfe** [comment]
Command selectors will not be in 2.16.0.

**Comment by smmmadden** [comment]
Hi @md678685 - could you add Permission Node updates to this list?  There will be some that need to be deprecated for 1.13.x servers based on the list I took from https://essinfo.xeya.me/permissions.php

essentials.build.break.<id>[:datavalue]
essentials.build.craft.<id>[:datavalue]
essentials.build.drop.<id>[:datavalue]
essentials.build.interact.<id>[:datavalue]
essentials.build.pickup.<id>[:datavalue]
essentials.build.place.<id>[:datavalue]
essentials.itemspawn.item-[itemid]
essentials.unlimited.item-[itemid]

Not sure which are still valid or will/should be invalid as the list hasn't been updated in a while.  I did check each of the EssentialsX*.jar files for each of the plugin.yml and we're not listing them there for me to compare what's supported or not.  Would be much easier on users if they were self-contained in the jars.  I'm seeing many developers starting to do this with other plugins.

Full Permissions: https://pastebin.com/zUpxqbTf I've set the paste to never expire.
Thanks -Steve

**Comment by mdcfe** [comment]
@smmmadden I don't have any control over the unofficial permissions list page - that's run by @Xeyame.

Many plugins traditionally list permissions in the `plugin.yml`, but EssentialsX doesn't as registering them causes undesirable side-effects that interfere with EssentialsX permission handling, as well as creating an unnecessary maintenance headache.

**Comment by smmmadden** [comment]
@md678685 okay, so where is the official permissions list page and has it been updated?  I just wanted to make sure that we cover perms as well as block changes that are in flight. :-)

**Comment by mdcfe** [comment]
@smmmadden There is the Essentials permissions reference and the list provided by Xeyame as a convenience to the EssentialsX community. EssentialsX does not currently provide an official permissions reference.

**Comment by smmmadden** [comment]
@md678685 - thanks, no sense in recreating another document then if its pretty current.  So it comes down to what we're doing about these for the 1.13+ releases?
essentials.build.break.[:datavalue]
essentials.build.craft.[:datavalue]
essentials.build.drop.[:datavalue]
essentials.build.interact.[:datavalue]
essentials.build.pickup.[:datavalue]
essentials.build.place.[:datavalue]
essentials.itemspawn.item-[itemid]
essentials.unlimited.item-[itemid]
These are the only ones referring to data value and itemid's that are not valid for new releases (for everything anyway).
